plumbing/object/commit_walker.go

```diff
@@ -181,3 +182,107 @@ func (w *commitPostIterator) ForEach(cb func(*Commit) error) error {
 }
 
 func (w *commitPostIterator) Close() {}
+
+// commitAllIterator stands for commit iterator for all refs.
+type commitAllIterator struct {
+	// el points to the current element.
+	el *list.Element
+}
+
+// NewCommitAllRefsIter returns a new commit iterator for all refs.
+func NewCommitAllRefsIter(
+	storer storer.EncodedObjectStorer,
+	commitIterFunc func(*Commit) CommitIter,
+	commitIter CommitIter,
+	refIter storer.ReferenceIter) CommitIter {
+
+	l := list.New()
+	m := make(map[plumbing.Hash]*list.Element)
+
+	commitIter.ForEach(func(c *Commit) error {
```

Comment:
Maybe we don't need to populate a list by _first iterator_ (in other words for --all we may ignore `From` option and just go through all references)